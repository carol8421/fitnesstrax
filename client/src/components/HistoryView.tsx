import React from "react"
import moment from "moment"
import { connect } from "react-redux"

//import { HistoryEntry } from "../types"
import { isSomething } from "../common"
//import { runFetchHistory } from "../state/actions"
import { getCurrentlyEditing, getHistory, getRange } from "../state/state"

import { DailyEntryView, DailyEntryEditView } from "./DailyEntry"

export interface Props {
  //fetchData: () => void
  history: any
  range: any
  currentEdit: any
}

class History extends React.Component<Props, object> {
  /*
  componentDidMount() {
    this.props.fetchData()
  }
  */

  render() {
    return <div />
    /*
    if (isSomething(this.props.history)) {
      const today = moment({ hour: 0, minute: 0, second: 0 })
      const [start, end] = isSomething(this.props.range)
        ? this.props.range
        : [this.props.history.startRange(), today]
      const days = fillInDayRange(start, end).reverse()
      return (
        <div>
          {days.map(date => {
            var entry = this.props.history.entry(date)
            if (
              isSomething(entry) &&
              isSomething(this.props.currentEdit) &&
              this.props.currentEdit.isSame(entry.date)
            ) {
              return <DailyEntryEditView date={date} data={entry} />
            } else if (
              isSomething(entry) &&
              isSomething(this.props.currentEdit) &&
              !this.props.currentEdit.isSame(entry.date)
            ) {
              return <DailyEntryView date={date} data={entry} />
            } else if (
              isSomething(entry) &&
              !isSomething(this.props.currentEdit)
            ) {
              return <DailyEntryView date={date} data={entry} />
            } else if (
              !isSomething(entry) &&
              isSomething(this.props.currentEdit) &&
              this.props.currentEdit.isSame(date)
            ) {
              return <DailyEntryEditView date={date} data={entry} />
            } else if (
              !isSomething(entry) &&
              isSomething(this.props.currentEdit) &&
              !this.props.currentEdit.isSame(date)
            ) {
              return (
                <DailyEntryView
                  date={date}
                  data={{ weightRecord: null, timeDistanceRecords: [] }}
                />
              )
            } else if (
              !isSomething(entry) &&
              !isSomething(this.props.currentEdit)
            ) {
              return (
                <DailyEntryView
                  date={date}
                  data={{ weightRecord: null, timeDistanceRecords: [] }}
                />
              )
            } else {
              throw new Error(
                `Unhandled case in History.render: ${entry}, ${
                  this.props.currentEdit
                }`,
              )
            }
          })}{" "}
        </div>
      )
    } else {
      return null
    }
    */
  }
}

/* *ew* There must be a better way to do this. I'm thinking a generator that successively applies a function to a vaule. */
const fillInDayRange = (startDate, endDate) => {
  var current = startDate.clone()
  var lst = [startDate]
  while (current.isBefore(endDate)) {
    current.add(1, "days")
    lst.push(current)
    current = current.clone()
  }
  return lst
}

export const HistoryView = connect(
  state => ({
    history: getHistory(state),
    currentEdit: getCurrentlyEditing(state),
    range: getRange(state),
  }),
  //(dispatch: any) => ({ fetchData: () => dispatch(runFetchHistory()) }),
)(History)
