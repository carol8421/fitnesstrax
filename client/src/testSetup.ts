import { createStore } from "redux"
import { IANAZone } from "luxon"

import Controller from "./controller"
import * as redux from "./redux"
import { English } from "./i18n"
import { USA, UserPreferences } from "./settings"

export const setupEnv = () => {
  const store = createStore(redux.rootReducer)
  const controller = new Controller("http://nowhere/", store)
  return {
    store,
    controller,
  }
}

export const standardPreferences = () =>
  new UserPreferences(IANAZone.create("America/New_York"), USA, English)
