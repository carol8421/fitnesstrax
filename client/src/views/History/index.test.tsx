import { mount } from "enzyme"
import moment from "moment"
import React from "react"
import { Provider } from "react-redux"

import { setupEnv } from "../../testSetup"
import HistoryView from "./index"

describe("HistoryView", () => {
  xit("renders when there is no history", () => {
    const { store, controller } = setupEnv()

    const wrapper = mount(
      <Provider store={store}>
        <HistoryView
          controller={controller}
          range={{
            start: moment("2017-10-23T22:09:00Z"),
            end: moment("2018-11-12T18:30:00Z"),
          }}
        />
      </Provider>,
    )

    console.log(wrapper.debug())
    expect(wrapper).toMatchSnapshot()
  })

  xit("renders all of the elements of a weights-only history", () => {
    /*
    const { store, controller } = setupEnv()
    store.dispatch(

    const wrapper = mount(
      <Provider store={store}>
        <HistoryView controller={controller} />
      </Provider>,
    )

    console.log(wrapper.debug())
    expect(wrapper).toMatchSnapshot()
  */
  })

  xit("renders reasonably with duplicate weight entries", () => {})

  xit("renders all of the elements of a timedistance-only history", () => {})

  xit("renders all of the elements of a timedistance-only history", () => {})
})
