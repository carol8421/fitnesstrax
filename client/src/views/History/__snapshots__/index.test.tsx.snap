// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryView renders when there is no history 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(History)
    controller={
      Controller {
        "authenticate": [Function],
        "client": Client {
          "appUrl": "http://nowhere/",
          "authenticate": [Function],
          "fetchHistory": [Function],
          "rfc3339Format": "YYYY-MM-DDTHH:mm:ssZ",
        },
        "fetchRecords": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "end": DateTimeTz {
                  "timestamp": "2018-11-12T18:30:00.000Z",
                },
                "start": DateTimeTz {
                  "timestamp": "2017-10-23T22:09:00.000Z",
                },
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "logout": [Function],
        "setPreferences": [Function],
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      }
    }
    range={
      Object {
        "end": DateTimeTz {
          "timestamp": "2018-11-12T18:30:00.000Z",
        },
        "start": DateTimeTz {
          "timestamp": "2017-10-23T22:09:00.000Z",
        },
      }
    }
  >
    <History
      controller={
        Controller {
          "authenticate": [Function],
          "client": Client {
            "appUrl": "http://nowhere/",
            "authenticate": [Function],
            "fetchHistory": [Function],
            "rfc3339Format": "YYYY-MM-DDTHH:mm:ssZ",
          },
          "fetchRecords": [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "end": DateTimeTz {
                    "timestamp": "2018-11-12T18:30:00.000Z",
                  },
                  "start": DateTimeTz {
                    "timestamp": "2017-10-23T22:09:00.000Z",
                  },
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "logout": [Function],
          "setPreferences": [Function],
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        }
      }
      dispatch={[Function]}
      history={Array []}
      prefs={
        UserPreferences {
          "language": Language {
            "name": "English",
            "sym": "en",
          },
          "timezone": IANAZone {
            "valid": true,
            "zoneName": "UTC",
          },
          "units": UnitSystem {
            "length": "km",
            "lengthRepr": undefined,
            "mass": "kg",
            "massRepr": undefined,
            "sym": "SI",
          },
          "withLanguage": [Function],
          "withTimezone": [Function],
          "withUnits": [Function],
        }
      }
      range={
        Object {
          "end": DateTimeTz {
            "timestamp": "2018-11-12T18:30:00.000Z",
          },
          "start": DateTimeTz {
            "timestamp": "2017-10-23T22:09:00.000Z",
          },
        }
      }
    >
      <div
        id="History"
      >
        <Range
          classes={Object {}}
          range={
            Object {
              "end": DateTimeTz {
                "timestamp": "2018-11-12T18:30:00.000Z",
              },
              "start": DateTimeTz {
                "timestamp": "2017-10-23T22:09:00.000Z",
              },
            }
          }
        >
          <div
            className=""
          >
            2017-10-23
             - 
            2018-11-12
          </div>
        </Range>
      </div>
    </History>
  </Connect(History)>
</Provider>
`;
